servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -552,6 +564,7 @@ public boolean equals(Object o) {
   private TraversalResult traverseToInode(String[] pathComponents, boolean collectNonPersisted)
       throws InvalidPathException {
     List<Inode> nonPersistedInodes = Lists.newArrayList();
+    List<Inode> inodesIncludingNull = Lists.newArrayList();
```

Comment:
I don't think we should be adding the "includingNull" because it makes the scope of possible uses of this method narrower ... instead, simply collect the inodes encountered on during the traversal ... the extra nulls can be created by the caller based on the index and the number of path components